### PR TITLE
Update go, dependencies and tools

### PR DESCRIPTION
**Pull request by clobrano**
- Update Go to version 1.21
- Update to k8s 29.1 and Openshift release-4.16 dependencies
- Update Dockerfile
- More dependency updates
- Update tools and update code and manifests

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -4,25 +4,25 @@ SHELL := /bin/bash
 OPERATOR_NAME := machine-deletion-remediation
 
 ## Tool versions
-# OPERATOR_SDK versions at github.com/operator-framework/operator-sdk/releases
-OPERATOR_SDK_VERSION ?= v1.30.0
+# OPERATOR_SDK versions at https://github.com/operator-framework/operator-sdk/releases
+OPERATOR_SDK_VERSION ?= v1.32.0
```
Comment (slintes) [inline Makefile:8]: any reason for not using 1.33?

File: bundle/manifests/machine-deletion-remediation.clusterserviceversion.yaml [hunk 2]
```diff
@@ -47,13 +47,13 @@ metadata:
     capabilities: Basic Install
     categories: OpenShift Optional
     containerImage: ""
-    createdAt: ""
+    createdAt: "2024-01-25T10:30:34Z"
```
Comment (slintes) [inline bundle/manifests/machine-deletion-remediation.clusterserviceversion.yaml:50]: this was empty before?

File: Makefile (hunk 1 above)
Comment (clobrano) [reply Makefile:8]: I'd like to go through the breaking changes after the release

File: bundle/manifests/machine-deletion-remediation.clusterserviceversion.yaml (hunk 2 above)
Comment (clobrano) [reply bundle/manifests/machine-deletion-remediation.clusterserviceversion.yaml:50]: ah damn

File: Makefile (hunk 1 above)
Comment (slintes) [reply Makefile:8]: There are breaking changes? Didn't notice that... 🙈 😄 But ok of course, just wanted to be sure that it's on purpose

**Comment by clobrano** [comment]
Ignoring codacity remarks on vendor and merge

**Pull request merged**